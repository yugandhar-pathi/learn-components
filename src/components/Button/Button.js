import PropTypes from 'prop-types'
import {StyledButton} from './Button.styles'

export const Button = ({label, dataTestId, primary, secondary, ...props}) => {
    return <StyledButton data-testid={dataTestId} primary={primary} secondary={secondary} {...props}>{label}</StyledButton>
}

Button.propTypes = {
    dataTestId: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    primary: PropTypes.bool,
    secondary: PropTypes.bool,
    size: PropTypes.oneOf(["small", "medium", "large"])
}

Button.defaultProps = {
    primary: false,
    secondary: false,
    size: "small"
}
