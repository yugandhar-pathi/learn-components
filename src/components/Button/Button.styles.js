import styled from 'styled-components'

export const StyledButton = styled.button`
 padding: ${(props) => props.secondary ? "8px" : "5px"};
 border-radius: '5px'; 
 background-color:${(props)=> props.primary ? "green" : ""}
`