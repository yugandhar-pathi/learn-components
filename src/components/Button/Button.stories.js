import React from 'react'
import { Button } from './Button';

export default {
    component: Button,
};

const Template = (args) => <Button {...args} />;

//export const Primary = () => <Button label="Button" primary />
export const Primary = Template.bind({});
Primary.args = {
    label: 'Button',
    primary: true
};

export const Secondary = Template.bind({});
Secondary.args = {
    label: 'Button',
    secondary: true
};