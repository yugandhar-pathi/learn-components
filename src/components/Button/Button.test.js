import {render, screen} from '@testing-library/react'
import { Button } from './Button'

test('Button component', () => {
    render(<Button label="test" dataTestId="test-button" />)

    expect(screen.getByTestId('test-button').textContent).toBe("test")
})

test('Primary button Snapshot', () => {
    render(<Button label="primary" primary dataTestId="test-button" />)

    expect(screen.getByTestId('test-button')).toMatchSnapshot()
})

test('Secondary button Snapshot', () => {
    render(<Button label="secondary" secondary dataTestId="test-button" />)

    expect(screen.getByTestId('test-button')).toMatchSnapshot()
})