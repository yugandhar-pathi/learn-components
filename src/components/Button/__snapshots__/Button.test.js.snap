// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary button Snapshot 1`] = `
<button
  class="sc-bczRLJ ipkIqN"
  data-testid="test-button"
>
  primary
</button>
`;

exports[`Secondary button Snapshot 1`] = `
<button
  class="sc-bczRLJ gOcKSH"
  data-testid="test-button"
>
  secondary
</button>
`;
